src/main/java/zeenea/connector/common/IdentificationProperty.java

```diff
@@ -0,0 +1,68 @@
+package zeenea.connector.common;
+
+import static zeenea.connector.exception.ExceptionUtils.throwIfNullOrEmpty;
+
+import java.util.Objects;
+import java.util.StringJoiner;
+import org.jetbrains.annotations.NotNull;
+
+public class IdentificationProperty {
+  @NotNull private final String key;
+  @NotNull private final String value;
+
+  private IdentificationProperty(Builder builder) {
+    this.key = builder.key;
+    this.value = builder.value;
+  }
+
+  public @NotNull String getKey() {
+    return key;
+  }
+
+  public @NotNull String getValue() {
+    return value;
+  }
+
+  public static Builder builder(@NotNull String key, @NotNull String value) {
+    return new Builder(key, value);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    IdentificationProperty that = (IdentificationProperty) o;
+    return Objects.equals(key, that.key) && Objects.equals(value, that.value);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(key, value);
+  }
+
+  @Override
+  public String toString() {
+    return new StringJoiner(", ", IdentificationProperty.class.getSimpleName() + "[", "]")
+        .add("key='" + key + "'")
+        .add("value='" + value + "'")
+        .toString();
+  }
+
+  public static class Builder {
```

Comment:
Ce builder n'a aucun sens. Il n'apporte rien par rapport à une factory à deux paramètres qui renvoie directement l'objet construit.

Reply:
Nous avons gardé le builder en l'état mais nous avons rajouté une factory `IdentificationProperty of(@NotNull String key, @NotNull String value)` pour faciliter son instanciation.